components/elasticsearch/scripts/create.py

```diff
@@ -262,14 +267,96 @@ def main():
                         'index already exists...')
 
         if http_request('http://{0}:{1}/cloudify_storage'.format(
-                es_endpoint_ip, es_endpoint_port), method='HEAD'):
+                es_endpoint_ip, es_endpoint_port), method='HEAD').code == 200:
             utils.error_exit('\'cloudify_storage\' index already exists on '
                              '{0}, terminating bootstrap...'.format(
                                  es_endpoint_ip))
         _configure_elasticsearch(host=es_endpoint_ip, port=es_endpoint_port)
 
+    if utils.is_upgrade:
+        restore_upgrade_data(es_endpoint_ip, es_endpoint_port)
+
+    if not es_endpoint_port:
+        utils.systemd.stop(ES_SERVICE_NAME, append_prefix=False)
+
     ctx.instance.runtime_properties['es_endpoint_ip'] = es_endpoint_ip
 
 
-if utils.is_install:
-    main()
+def _get_es_install_port():
+    es_props = utils.ctx_factory.get(ES_SERVICE_NAME, upgrade_props=False)
+    return es_props['es_endpoint_port']
+
+
+def _get_es_install_endpoint():
+    es_props = utils.ctx_factory.get(ES_SERVICE_NAME, upgrade_props=False)
+    if es_props['es_endpoint_ip']:
+        es_endpoint = es_props['es_endpoint_ip']
+    else:
+        es_endpoint = ctx.instance.host_ip
+    return es_endpoint
+
+
+def dump_upgrade_data():
+
```

Comment:
That did come up, but was not decided. This solution would require some changes to the snapshot API and after discussing with Dan and considering the time constraints, we thought this should be fine. Is there a specific reason why you think we should use the snapshots API.